compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala

```diff
@@ -694,6 +655,77 @@ abstract class PrepJSInterop extends plugins.PluginComponent
       enterOwner(kind) { super.transform(implDef) }
     }
 
+    private def checkAndComputeJSNativeLoadSpecOf(pos: Position,
+        sym: Symbol): JSNativeLoadSpec = {
+      if (enclosingOwner is OwnerKind.JSNativeMod) {
+        for {
+          annot <- sym.annotations
+          annotSym = annot.symbol
+          if annotSym != JSNameAnnotation && JSNativeLoadingSpecAnnots.contains(annotSym)
+        } {
+          reporter.error(annot.pos,
+              "Classes and objects nested in a JS native object cannot have " +
+              s"an ${annotSym.nameString} annotation.")
+        }
+
+        val ownerLoadSpec = jsInterop.jsNativeLoadSpecOf(sym.owner)
+        val jsName = jsInterop.jsNameOf(sym)
+        ownerLoadSpec match {
+          case JSNativeLoadSpec.Global(path) =>
+            JSNativeLoadSpec.Global(path :+ jsName)
+          case JSNativeLoadSpec.Import(module, path) =>
+            JSNativeLoadSpec.Import(module, path :+ jsName)
+        }
+      } else {
+        val loadSpecAnnot =
+          checkAndGetJSNativeLoadingSpecAnnotOf(pos, sym)
+
+        loadSpecAnnot match {
+          case Some(annot) if annot.symbol == JSGlobalScopeAnnotation =>
+            if (!sym.isModuleClass) {
+              reporter.error(pos,
+                  "Only native JS objects can have an " +
+                  "@JSGlobalScope annotation (or extend js.GlobalScope).")
+            }
+            JSNativeLoadSpec.Global(Nil)
+
+          case Some(annot) if annot.symbol == JSImportAnnotation =>
+            val module = annot.stringArg(0).getOrElse {
+              "" // do not care because it does not compile anyway
+            }
+            annot.stringArg(1).fold {
+              JSNativeLoadSpec.Import(module, Nil)
+            } { pathName =>
+              val path = pathName.split('.').toList
+              JSNativeLoadSpec.Import(module, path)
+            }
+
+          case _ if loadSpecAnnot.forall(_.symbol == JSNameAnnotation) =>
```

Comment:
I'm not sure this is very clear and worth the effort just to not duplicate the creation of `JSNativeLoadSpec.Global(path)`. Consider:

```scala
def globalFromName = {
  val path = jsInterop.jsNameOf(sym).split('.').toList
  JSNativeLoadSpec.Global(path)
}

checkAndGetJSNativeLoadingSpecAnnotOf(sym) match {
  // snip

  case Some(annot) if annot.symbol == JSNameAnnotation =>
    globalFromName

  case None =>
    val needsExplicitJSName = ...
    // etc.

    globalFromName
```

Factor